ci/base.Dockerfile

```diff
@@ -24,6 +24,7 @@ RUN apt-get update -qq && apt-get install -qq -y --no-install-recommends \
     python-openssl \
     libreadline-dev \
     git \
+    rustc \
```

Comment:
Out of curiosity: why is `rust` needed?

Reply:
Previously, the CI failed at https://gitlab.com/cscs-ci/ci-testing/webhook-ci/mirrors/4455690602105886/4525297225819146/-/jobs/3744176151

This is an excerpt of the output:

> error: subprocess-exited-with-error
>  
>  × Preparing metadata (pyproject.toml) did not run successfully.
  │ exit code: 1
  ╰─> [6 lines of output]
>      
>      Cargo, the Rust package manager, is not installed or is not on PATH.
>      This package requires Rust and Cargo to compile extensions. Install it through
>      the system's package manager or via https://rustup.rs/
>      
>      Checking for Rust toolchain....
>      [end of output]

As much as I could find out there is a dependency of a dependency that causes this problem whenever a version is requested for which there is no wheel available (yet). It might have something to do with the python package `orjson` which one or multiple of our dependencies depend on in turn.